network/network.go

```diff
@@ -7,10 +7,13 @@ import (
 	"github.com/ava-labs/avalanche-network-runner/network/node"
 )
 
+var ErrUndefined = errors.New("undefined network")
 var ErrStopped = errors.New("network stopped")
 
 // Network is an abstraction of an Avalanche network
 type Network interface {
+	// Load network config
+	LoadConfig(context.Context, Config) error
```

Comment:
It seems like it's require to call this right after the network is created...if that's so, it should just be part of `NewNetwork`. Also, can we please add a more detailed specification?

Reply:
After network instance creation, you can also configure and start the network from a given snapshot. So prefered to let network instance creation to do only fields initialization.
Related reasons:
1) it seems more uniform to add the network config input param to the interface, considering that AddNode also takes a node config
2) snapshot load implementation is based on network config load, and preferred to add all snapshot management functions to the interface